mrpeek: added no text mode, modified boolean cmd line options

maxpietsch opened the pull request:
Added an option to remove text output in interactive or non-interactive mode. In non-interactive mode, this allows writing a sixel image to file via 

```
mrpeek im.mif -text no -inter no > im.sixel
mrpeek im.mif -text no -inter no -image no -cross -1 -1 -plot 0 > plot.sixel
``` 

which can then be displayed via `cat` on supported terminals or converted to png using [sixel2png](img2sixel).

maxpietsch committed:
mrpeek: added no text mode (key [t] and cmd line option)

maxpietsch committed:
mrpeek: added command line option to disable main image

jdtournier commented on cmd/mrpeek.cpp:
```diff
@@ -580,23 +588,26 @@ std::string display (ImageType& image, Sixel::ColourMaps& colourmaps)
+  if (show_text) out += show_focus(image);
 
-  if (orthoview) {
+  if (interactive and orthoview and show_text) {
```
Been writing a lot of python lately...?

maxpietsch replied:
```c++
%:include <iostream>
 
struct X
<%
    compl X() <%%> // destructor
    X() <%%>
    X(const X bitand) = delete; // copy constructor
 
    bool operator not_eq(const X bitand other)
    <%
       return this not_eq bitand other;
    %>
%>;
 
int main(int argc, char* argv<::>) 
<%
    // lambda with reference-capture:
    auto say = <:bitand:>(const char* name)
    <%
        std::cout << "I like " << name << '\n';
    %>;
 
    if (argc > 1 and argv<:1:> not_eq nullptr) <%
        say(argv<:1:>);
    %>
    else <%
        say("Python");
    %>
%>
```

jdtournier replied:
Wow, I had no idea there was such a thing as [the alternative operators](https://en.cppreference.com/w/cpp/language/operator_alternative)... Guess you learn something new every day!

jdtournier commented:
Looks good, other than some slightly iffy syntax in places... ;)

Couple of thoughts, not necessarily related to this PR as such:

- I reckon it would be nice to also default to non-interactive mode if `stdout` is not pointing to a TTY. We could use the same approach we already use on MRtrix core to disable vt100 codes and switch the progressbar behaviour when redirected to file.

- every time I have to use the `-interactive` option, it bothers me that we have to specify 0 or no when clearly the default mode of action is interactive. I'd prefer to provide a single `-batch` option with no  additional arguments. 

  Similar with the `-orthoview` option. I think I'd rather reinterpret the `-axis` option as requesting non-orthoview mode with the projection requested. Could also replace that `-axis` option with (or just add) more explicit `-axial`, `-sagittal` and `-coronal` options, again with the understanding that this disables orthoview.

maxpietsch committed:
mrpeek: remove alternative tokens

maxpietsch committed:
mrpeek: simplify options

- removed axis and orthoview options.
- interactive y/n --> batch
- added sagittal, coronal, axial options which deactivate orthoview
- image y/n and text y/n --> noimage notext

maxpietsch commented:
Simplified the options as suggested and changed `-image y/n` and `-text y/n` to `-noimage` and `-notext` in e9dcfda. The TTY detection is not implemented yet.

```
OPTIONS

  -batch
     disables interactive mode

  -sagittal
     view sagittal projection only. Default: orthoview

  -coronal
     view coronal projection only. Default: orthoview

  -axial
     view axial projection only. Default: orthoview

  -slice number
     select slice to display

  -plot number
     specify plot dimension: 0: L/R (sagittal); 1: A/P (coronal); 2 I/S
     (axial); 3 volumes...

  -intensity_range min max
     specify intensity range of the data. The image intensity will be scaled
     between the specified minimum and maximum intensity values. By default,
     percentile scaling is used.

  -percentile_range min max
     specify intensity range of the data. The image intensity will be scaled
     between the specified minimum and maximum percentile values. Defaults are:
     0.2 - 99.8

  -colourmap name
     the colourmap to apply; choices are:
     gray,hot,cool,jet,inferno,viridis,pet. Default is gray.

  -crosshairs x y
     draw crosshairs at specified position. Set to negative position to hide.

  -levels number
     number of intensity levels in the colourmap. Default is 32.

  -zoom factor
     scale the image size by the supplied factor

  -notext
     omit text output to show only the sixel image

  -noimage
     do not render the main image
```

jdtournier committed:
mrpeek: use non-interactive mode if stdout is not TTY

jdtournier committed:
mrpeek: alternate -focus option to set focus position

and -nocrosshairs option to hide crosshairs

jdtournier commented:
5f78ff7 should address most of the issues in #39 - apart from the `-scanner` option.

maxpietsch committed:
mrpeek: focus is explicitly integer valued

maxpietsch commented:
I made the focus explicitly integer-valued otherwise floating-point values would just be truncated. Looks all good to me.

jdtournier commented:
Nope, has to be floating point if you want to be able to skip indices, as per the example in the command line usage. An empty entry returns a NaN for a sequence of floats, but that's not supported for a sequence of ints.

maxpietsch committed:
Revert "mrpeek: focus is explicitly integer valued"

This reverts commit fb17b8938ebd462fedd175d45dfcaa9cbe0a8a6b.

maxpietsch committed:
mrpeek: focus: round instead of truncate to integer

maxpietsch commented:
Ok, but I'd round instead of truncate: c3fddd2

jdtournier commented:
👍
